pkg/operator/ceph/object/bucket/predicate.go

```diff
@@ -62,8 +64,18 @@ func predicateController(ctx context.Context, c client.Client) predicate.Funcs {
 						}
 					}
 				}
-			}
 
+			case *cephv1.CephObjectStore:
+				if objOld, ok := e.ObjectOld.(*cephv1.CephObjectStore); ok {
+					if objNew, ok := e.ObjectNew.(*cephv1.CephObjectStore); ok {
+						// if objOld.Spec.Gateway.ExternalRgwEndpoints[0] != objNew.Spec.Gateway.ExternalRgwEndpoints[0] {
+						logger.Infof("there is an update event on CephObjectStore CR, so reconcile the bucket provisioner %s %s", objOld.Spec.Gateway.ExternalRgwEndpoints[0].IP, objNew.Spec.Gateway.ExternalRgwEndpoints[0].IP)
```

Comment:
I see the old and new objects as same and they both are updated `so reconcile the bucket provisioner 10.1.115.107 10.1.115.107`